src/virtual-machine.js

```diff
@@ -447,6 +448,14 @@ class VirtualMachine extends EventEmitter {
     }
 
     /**
+     * Handle a Blockly event for the flyout.
+     * @param {!Blockly.Event} e Any Blockly event.
+     */
+    variableListener (e) {
+        this.runtime.blocklyListen(e);
```

Comment:
You may want to filter for only variable events similar to monitorBlockListener() above.